packages/amplify-category-auth/resources/lambda-function.js

```diff
@@ -6,12 +6,43 @@ exports.handler = async function (event, context) {
     const userPoolId = event.ResourceProperties.userpoolId;
     const lambdaConfig = event.ResourceProperties.lambdaConfig;
     const config = {};
+    const cognitoClient = new aws.CognitoIdentityServiceProvider();
+    const userPoolConfig = await cognitoClient.describeUserPool({ UserPoolId: userPoolId }).promise();
+    const userPoolParams = userPoolConfig.UserPool;
+    // update userPool params
+
+    const updateUserPoolConfig = {
+      UserPoolId: userPoolParams.Id,
+      Policies: userPoolParams.Policies,
+      SmsVerificationMessage: userPoolParams.SmsVerificationMessage,
+      AccountRecoverySetting: userPoolParams.AccountRecoverySetting,
+      AdminCreateUserConfig: userPoolParams.AdminCreateUserConfig,
+      AutoVerifiedAttributes: userPoolParams.AutoVerifiedAttributes,
+      EmailConfiguration: userPoolParams.EmailConfiguration,
+      EmailVerificationMessage: userPoolParams.EmailVerificationMessage,
+      EmailVerificationSubject: userPoolParams.EmailVerificationSubject,
+      VerificationMessageTemplate: userPoolParams.VerificationMessageTemplate,
+      SmsAuthenticationMessage: userPoolParams.SmsAuthenticationMessage,
+      MfaConfiguration: userPoolParams.MfaConfiguration,
+      DeviceConfiguration: userPoolParams.DeviceConfiguration,
+      SmsConfiguration: userPoolParams.SmsConfiguration,
+      UserPoolTags: userPoolParams.UserPoolTags,
+      UserPoolAddOns: userPoolParams.UserPoolAddOns,
+    };
+
+    // removing undefined keys
+    Object.keys(updateUserPoolConfig).forEach(key => updateUserPoolConfig[key] === undefined && delete updateUserPoolConfig[key]);
+
+    /*removing UnusedAccountValidityDays as deprecated
+    InvalidParameterException: Please use TemporaryPasswordValidityDays in PasswordPolicy instead of UnusedAccountValidityDays
+    */
+    delete userPoolParams.AdminCreateUserConfig.UnusedAccountValidityDays;
```

Comment:
`userPoolParams.AdminCreateUserConfig` is always defined, SDK docs says optional? what if previous delete removed the key?

Add defensive coding here

Reply:
done